hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoApproveAllowanceTransitionLogic.java

```diff
@@ -0,0 +1,212 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+package com.hedera.services.txns.crypto;
+
+import com.hedera.services.context.TransactionContext;
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.ledger.SigImpactHistorian;
+import com.hedera.services.state.submerkle.FcTokenAllowance;
+import com.hedera.services.state.submerkle.FcTokenAllowanceId;
+import com.hedera.services.store.AccountStore;
+import com.hedera.services.store.models.Account;
+import com.hedera.services.store.models.Id;
+import com.hedera.services.txns.TransitionLogic;
+import com.hedera.services.txns.crypto.validators.AllowanceChecks;
+import com.hedera.services.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.CryptoAllowance;
+import com.hederahashgraph.api.proto.java.NftAllowance;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenAllowance;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateFalse;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.MAX_ALLOWANCES_EXCEEDED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+
+public class CryptoApproveAllowanceTransitionLogic implements TransitionLogic {
+	private final TransactionContext txnCtx;
+	private final SigImpactHistorian sigImpactHistorian;
+	private final AccountStore accountStore;
+	private final AllowanceChecks allowanceChecks;
+	private final GlobalDynamicProperties dynamicProperties;
+
+	@Inject
+	public CryptoApproveAllowanceTransitionLogic(
+			final TransactionContext txnCtx,
+			final SigImpactHistorian sigImpactHistorian,
+			final AccountStore accountStore,
+			final AllowanceChecks allowanceChecks,
+			final GlobalDynamicProperties dynamicProperties) {
+		this.txnCtx = txnCtx;
+		this.sigImpactHistorian = sigImpactHistorian;
+		this.accountStore = accountStore;
+		this.allowanceChecks = allowanceChecks;
+		this.dynamicProperties = dynamicProperties;
+	}
+
+	@Override
+	public void doStateTransition() {
+		/* --- Extract gRPC --- */
+		final TransactionBody cryptoApproveAllowanceTxn = txnCtx.accessor().getTxn();
+		final AccountID owner = cryptoApproveAllowanceTxn.getTransactionID().getAccountID();
```

Comment:
I think we should use the field in the new protobuf messages. And in sig reqs we should validate if the ownerId specified in each allowance has signed the txn or not. 
Although on paper this owner ID and payer id are same.. we have to validate they are and fail appropriately

Reply:
@anighanta  We do check if the payer is same as owner. Or else return `PAYER_AND_OWNER_NOT_EQUAL` for each allowances. The check is [here](https://github.com/hashgraph/hedera-services/blob/0ab22dce41fa3c8ff65a2a24ee8f74ca3fcc0ade/hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AllowanceChecks.java#L137)